pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -281,11 +309,18 @@ func (ds *dockerService) getIP(sandbox *dockertypes.ContainerJSON) (string, erro
 		// reporting the IP.
 		return "", nil
 	}
-	if IP, err := ds.getIPFromPlugin(sandbox); err != nil {
-		glog.Warningf("%v", err)
-	} else if IP != "" {
-		return IP, nil
+
+	// Don't bother getting IP if the pod is known and networking isn't ready
+	ready, ok := ds.getNetworkReady(podSandboxID)
+	if ok && !ready {
+		return "", nil
 	}
+
+	ip, err := ds.getIPFromPlugin(sandbox)
+	if err == nil {
```

Comment:
@freehan no, that's mostly the point of this PR.  We don't want to warn in *this* function, but we want to pass the warning up to higher levels, eg the caller of PodSandboxStatus() so it can do whatever it wants with that information, whether it's warn or ignore.